### ISSUE TITLE
filebeat error and excessive memory usage with autodiscover

### ISSUE THREAD

**Issue by bwunderlich824** [elastic/beats#9078]
This is a new ticket based on the closed but on going issues noted here (https://github.com/elastic/beats/issues/6503) filebeat logs show several errors and memory usage grows until the pod is shut down by kubernetes. I am using filebeat v6.4.3. 

filebeat logs show the following errors:
2018-11-14T16:50:43.002Z	ERROR	kubernetes/watcher.go:254	kubernetes: Watching API error EOF
2018-11-14T16:50:43.004Z	ERROR	kubernetes/watcher.go:254	kubernetes: Watching API error proto: wrong wireType = 6 for field ServiceAccountName

My configuration:
```
    setup.template.enabled: true
    setup.dashboards.enabled: false

    #Kubernetes AutoDiscover
    filebeat.autodiscover:
      providers:
        - type: kubernetes
          templates:

            #JSON LOGS
            - condition:
                equals:
                  kubernetes.labels.json_logs: "true"
              config:
                - type: docker
                  containers.ids:
                    - "${data.kubernetes.container.id}"
                  json.keys_under_root: true
                  json.add_error_key: true
                  processors:
                    - add_kubernetes_metadata:
                        in_cluster: true
                    - drop_fields:
                        fields: ["OriginContentSize","Overhead","BackendURL.Fragment","BackendURL.Scheme","ClientUsername","source","request_Cookie","request_Proxy-Authenticate","downstream_X-Authentication-Jwt","downstream_Set-Cookie","origin_Set-Cookie","downstream_Cache-Control"]

            #Non-JSON logs
            - condition:
                not:
                  equals:
                    kubernetes.labels.json_logs: "true"
              config:
                - type: docker
                  containers.ids:
                    - "${data.kubernetes.container.id}"
                  processors:
                    - add_kubernetes_metadata:
                        in_cluster: true

    cloud.id: ${ELASTIC_CLOUD_ID}
    cloud.auth: ${ELASTIC_CLOUD_AUTH}

    output.elasticsearch:
      hosts: ['<ELASTICSEARCH>:443']
      username: ${ELASTICSEARCH_USERNAME}
      password: ${ELASTICSEARCH_PASSWORD}
    tags: ["<ENV>"]
    logging.level: warning
    logging.json: false
```

**Comment by exekias** [comment]
Thanks for opening this @bwunderlich824, could you share a broader log? I'm particularly interested in other messages, errors and timings.

Best regards

**Comment by bwunderlich824** [comment]
Not alot else in there besides what was posted above. 
[logs.txt](https://github.com/elastic/beats/files/2617840/logs.txt)

**Comment by gamer22026** [comment]
This seems to be the sequence every time the **Watching API error** hits:
```
2018-12-04T22:06:43.754Z        ERROR   kubernetes/watcher.go:254       kubernetes: Watching API error EOF
2018-12-04T22:06:43.754Z        INFO    kubernetes/watcher.go:238       kubernetes: Watching API for resource events
2018-12-04T22:06:43.758Z        INFO    input/input.go:149      input ticker stopped
2018-12-04T22:06:43.758Z        INFO    input/input.go:167      Stopping Input: 7990041433892801910
2018-12-04T22:06:43.758Z        INFO    log/harvester.go:275    Reader was closed: /var/lib/docker/containers/8e7cba8ad5631f38c62c2fdbeac1f73b4987ff69f4a7a3521b2d0ead35b5b268/8e7cba8ad5631f38c62c2fdbeac1f73b4987ff69f4a7a3521b2d0ead35b5b268-json.log. Closing.
2018-12-04T22:06:43.758Z        ERROR   [autodiscover]  cfgfile/list.go:96      Error creating runner from config: Can only start an input when all related states are finished: {Id:7870214-51713 Finished:false Fileinfo:0xc42045b1e0 Source:/var/lib/docker/containers/8e7cba8ad5631f38c62c2fdbeac1f73b4987ff69f4a7a3521b2d0ead35b5b268/8e7cba8ad5631f38c62c2fdbeac1f73b4987ff69f4a7a3521b2d0ead35b5b268-json.log Offset:230605 Timestamp:2018-12-04 22:06:20.083047508 +0000 UTC m=+4508.847580429 TTL:-1ns Type:docker Meta:map[] FileStateOS:7870214-51713}
2018-12-04T22:06:43.759Z        INFO    log/input.go:138        Configured paths: [/var/lib/docker/containers/8e7cba8ad5631f38c62c2fdbeac1f73b4987ff69f4a7a3521b2d0ead35b5b268/*.log]
2018-12-04T22:06:43.759Z        INFO    input/input.go:114      Starting input of type: docker; ID: 7990041433892801910
2018-12-04T22:06:53.760Z        INFO    log/harvester.go:254    Harvester started for file: /var/lib/docker/containers/8e7cba8ad5631f38c62c2fdbeac1f73b4987ff69f4a7a3521b2d0ead35b5b268/8e7cba8ad5631f38c62c2fdbeac1f73b4987ff69f4a7a3521b2d0ead35b5b268-json.log
```

**Comment by bwunderlich824** [comment]
Issues still persists with filebeat 6.5.4

**Comment by bwunderlich824** [comment]
I pushed that version out to our development environment and I've had the same issue, the kubernetes pod just sucks up more memory until it reboots. Same logs as before. 

[logs.txt](https://github.com/elastic/beats/files/2741940/logs.txt)

**Comment by olivierboudet** [comment]
Same error here with 7.0.0-alpha2 and this configuration : 

```
filebeat:
      config:
        modules:
          - module: nginx
            access:
              enabled: true
            error:
              enabled: true
      autodiscover:
        providers:
          - type: kubernetes
            templates:
              - condition:
                  equals:
                    kubernetes.labels.type: "java"
                config:
                  - type: docker
                    containers.ids:
                      - "${data.kubernetes.container.id}"
                    multiline.pattern: '^\d{4}-\d{2}-\d{2} ' 
                    multiline.negate: true 
                    multiline.match: after
              - config:
                  - type: docker
                    containers.ids:
                      - "${data.kubernetes.container.id}"
                    processors: 
                      - add_kubernetes_metadata: 
                          in_cluster: true 
    setup:
      template:
        name: "filebeat"
        pattern: "filebeat-*"
    
    output.elasticsearch:
      hosts: ['${ELASTICSEARCH_HOST:elasticsearch}:${ELASTICSEARCH_PORT:9200}']  
      index: "%{[kubernetes.namespace]:nonamespace}-filebeat-%{+yyyy.MM.dd}"
      pipelines:
        - pipeline: java-logs-pipeline
          when.equals:
            kubernetes.labels.type: java
        - pipeline: mongodb-logs-pipeline
          when.equals:
            kubernetes.labels.app: mongo-pod
```

**Comment by cnelson** [comment]
I see a similar issue on my kubernetes clusters, filebeat will continue to use memory until exhausted, logging the messages described by @gamer22026.

It's a fairly linear leak, I don't see any huge steps in usage:

![image](https://user-images.githubusercontent.com/604163/52367725-9551a300-2a01-11e9-85de-3f128c8b7068.png)

**Comment by ruflin** [comment]
@cnelson Which version of filebeat are you using?

**Comment by bwunderlich824** [comment]
I just tried with the latest 6.6.0 and Have the samee issues. 
<img width="1681" alt="screen shot 2019-02-11 at 11 24 03 am" src="https://user-images.githubusercontent.com/30732263/52584616-be9e7480-2def-11e9-8707-c793e4fcfa94.png">

**Comment by kovetskiy** [comment]
The issue persists with `7.0.0-alpha2` too.

**Comment by JCMais** [comment]
I'm having the same issue, and having the same messages being logged by filebeat, plus some other ones.

I've posted more details on this other issue here: https://github.com/elastic/beats/issues/9302#issuecomment-471120645

**Comment by bwunderlich824** [comment]
Upgraded to v7.0.1 and still having the same issues.

**Comment by exekias** [comment]
this should be now fixed, more details can be found here: #9302
